### PR TITLE
Added the code of max dot product of two subsequence in the dp section 

### PR DESCRIPTION
**Pull request by YashKesh**
- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

### CODE REVIEW

**Review by Chulo1234zz — APPROVED** [review]

File: src/main/java/com/thealgorithms/backtracking/combinationSum.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+     * Recursively find combinations that sum to the target.
+     *
+     * @param res       The list to store valid combinations.
+     * @param candidates The array of candidate numbers.
+     * @param target    The target sum to achieve.
+     * @param sum1      The current sum during the recursion.
+     * @param start     The starting index for candidates.
+     * @param list1     The current combination being constructed.
+     */
+    public void check(List<List<Integer>> res, int[] candidates, int target, int sum1, int start, List<Integer> list1) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/combinationSum.java:22]: Is it the same as https://github.com/TheAlgorithms/Java/blob/574138c7a35351a0837bb4bd56e2eb295064b690/src/main/java/com/thealgorithms/dynamicprogramming/Sum_Of_Subset.java#L3?

Comment (YashKesh) [reply src/main/java/com/thealgorithms/backtracking/combinationSum.java:22]: No both of these questions are different @siriak

File: src/main/java/com/thealgorithms/backtracking/combinationSum.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+import java.util.*;
+
+/**
+ * Java program for solving the Combination Sum problem.
+ * Author: Yash Kesharwani
+ * File: CombinationSum.java
+ * Comments: This program finds all combinations of candidates that sum to the target.
+ */
+public class combinationSum {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/combinationSum.java:11]: Please use PascalCase for class names and camelCase for member names

File: src/main/java/com/thealgorithms/backtracking/combinationSum.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+     */
+    public List<List<Integer>> combinationsum(int[] candidates, int target) {
+        List<List<Integer>> res = new ArrayList<>();
+        int start = 0, sum1 = 0;
+        List<Integer> list1 = new ArrayList<>();
+        check(res, candidates, target, 0, 0, list1);
+        return res;
+    }
+
+    public static void main(String[] args) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/backtracking/combinationSum.java:53]: Please remove main and add some JUnit tests (see https://github.com/TheAlgorithms/Java/tree/master/src/test/java/com/thealgorithms)

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/backtracking/combinationSum.java (hunk 3 above)
Comment (YashKesh) [reply src/main/java/com/thealgorithms/backtracking/combinationSum.java:53]: Okay

**Pull request closed without merging**